mon-pix/app/components/routes/register-form.js

```diff
@@ -56,10 +59,10 @@ export default Component.extend({
       || !isDayValid(this.dayOfBirth) || !isMonthValid(this.monthOfBirth) || !isYearValid(this.yearOfBirth);
   }),
 
-  isCreationFormNotValid: computed('studentDependentUser.{email,username,password}', function() {
-    const isPasswordNotValid = !isPasswordValid(this.get('studentDependentUser.password'));
-    const isUsernameNotValid = !isStringValid(this.get('studentDependentUser.username'));
-    const isEmailNotValid = !isEmailValid(this.get('studentDependentUser.email'));
+  isCreationFormNotValid: computed('email,username,password', function() {
```

Comment:
Ici tu utilises une computed sous la forme : 
`'email, username, password'` équivalent à : `email && username && password`,
 tu peux aussi la faire de façon :
 `'email', 'username', 'password'` elle sera équivalente : `email || username || password`. 

Je me demande si ici, ce n'est pas le second cas que l'on cherche.

Reply:
Merci pour cet apprentissage, je connaissais pas la différence. 
Du moment ou l'username est calculé et que les trois champs sont obligatoires, je n'ai pas vu d'effet de bord. Cela dit, je mettrais la deuxième syntaxe que tu proposes.

Du coup avec la syntaxe brace extension:
` obj{prop1,prop2} est equivalent à 'obj.prop1' , 'obj.prop2' `

Reply:
@VincentHardouin je ne comprends pas ce que tu veux dire avec "équivalent à : `email && username && password`" ? Ou `email || username || password` ? Il n'y a pas d'opération booléenne qui soit faite sur les valeurs de ces propriétés à ma connaissance, je ne vois donc pas bien ce que viennent faire les opérateurs `&&` et `||` ici ?

Quant à la syntaxe des dépendances, à ma connaissance encore `computed('email,username,password', …)` ne fonctionne pas du tout au sens où j'imagine que ça va créer une dépendance à la propriété `email,username,password` qui n'existe évidemment pas. Je n'ai pas testé donc c'est possible que ça fasse quand même quelque chose mais en tout cas ce n'est pas documenté (tous les exemples de la doc sont écrits sous la forme `computed('email', 'username', 'password', …)`). Après il y a la `brace expansion` mais c'est encore autre chose.

Les dépendances de _computed properties_ dans Ember c'est le cauchemar à maintenir et surtout très difficile d'identifier quand on s'est trompé, souvent on a l'impression que ça marche alors qu'en fait la propriété ne va pas réagir quand une des dépendances va changer.

A priori maintenant qu'on est sous Octane on devrait pouvoir passer aux _tracked properties_ (https://blog.emberjs.com/2019/02/26/coming-soon-in-ember-octane-part-3.html) qui n'ont théoriquement pas ce problème ? @Radioreve  ?

Reply:
Pardon, je me suis mal exprimé, alors j'entends que d'écrire sous la forme `'email, username, password'` fonctionne si les trois valeurs changent (d'où dans ma tête le `email` et `username` et `password`), tandis que `'email', 'username', 'password'` recalcule seulement si une des 3 valeurs change.

Reply:
@VincentHardouin tu as déjà vu fonctionner une propriété avec `computed('email,username,password')` ? 😮

Reply:
@jonathanperret j'ai vu celle-[ci](https://github.com/1024pix/pix/blob/b0ebb25e19405929715dea05da25bbe22fc6d578/orga/app/controllers/authenticated/campaigns/list.js#L16)